x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,502 @@
+                  onChange={({ target }) =>
+                    onMaxBarsChange(Number(get(target, 'value', HISTOGRAM_MAX_BARS)))
+                  }
+                  placeholder={i18n.translate('xpack.lens.indexPattern.ranges.autoIntervals', {
+                    defaultMessage: 'Auto',
+                  })}
+                  showLabels
+                  showInput="inputWithPopover"
+                  prepend={i18n.translate('xpack.lens.indexPattern.ranges.maxIntervals', {
+                    defaultMessage: 'Max intervals',
+                  })}
+                />
+              ) : (
+                <EuiFieldNumber
+                  data-test-subj="lens-range-interval-field"
+                  value={interval}
+                  onChange={({ target }) => onIntervalChange(Number(target.value))}
+                  prepend={i18n.translate('xpack.lens.indexPattern.ranges.min', {
+                    defaultMessage: 'Min Interval',
+                  })}
+                  min={0}
+                />
+              )}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiButtonEmpty iconType="controlsHorizontal" onClick={() => onToggleEditor()}>
+            {i18n.translate('xpack.lens.indexPattern.ranges.customIntervalsToggle', {
+              defaultMessage: 'Create custom intervals',
+            })}
+          </EuiButtonEmpty>
+        </>
+      </EuiFormRow>
+    </>
+  );
+};
+
+const RangePopover = ({
+  range,
+  setRange,
+  Button,
+  isOpenByCreation,
+  setIsOpenByCreation,
+}: {
+  range: LocalRangeType;
+  setRange: (newRange: LocalRangeType) => void;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+  isOpenByCreation: boolean;
+  setIsOpenByCreation: (open: boolean) => void;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempRange, setTempRange] = useState(range);
+
+  const setPopoverOpen = (isOpen: boolean) => {
+    setIsPopoverOpen(isOpen);
+    setIsOpenByCreation(isOpen);
+  };
+
+  const saveRangeIfValid = (newRange: LocalRangeType, resetRange = false) => {
+    if (resetRange) {
+      // reset the temporary range for later use
+      setTempRange(range);
+    }
+    // send the range back to the main state
+    setRange(newRange);
+  };
+  const { from, to } = tempRange;
+
+  const safeLabel = getSafeLabel(tempRange);
+
+  const gteAppendLabel = i18n.translate('xpack.lens.indexPattern.ranges.greaterThanOrEqualAppend', {
+    defaultMessage: '\u2264',
+  });
+  const gteTooltipContent = i18n.translate(
+    'xpack.lens.indexPattern.ranges.greaterThanOrEqualTooltip',
+    {
+      defaultMessage: 'Greater than or equal to',
+    }
+  );
+  const ltPrependLabel = i18n.translate('xpack.lens.indexPattern.ranges.lessThanPrepend', {
+    defaultMessage: '\u003c',
+  });
+  const ltTooltipContent = i18n.translate('xpack.lens.indexPattern.ranges.lessThanTooltip', {
+    defaultMessage: 'Less than',
+  });
+
+  return (
+    <EuiPopover
```

Comment:
Please set `display="block"` on this popover, I think this is causing the `EuiLink` elements to not take up the full width and be hard to click.